jerry-core/jerry.c

```diff
@@ -2115,6 +2148,11 @@ jerry_value_to_boolean (const jerry_value_t value) /**< input value */
 {
   jerry_assert_api_available ();
 
+  if (ECMA_IS_VALUE_ERROR (value))
+  {
+    return false;
+  }
+
```

Comment:
Come to think of it, this is not a conversion, so even the internal api has a wrong name. Hence the value is not converted to boolen (e.g. a Boolean object is always true) so perhaps we need another conversion routine.

    var a = new Boolean(false); if (a) print("TRUE"); // yields true, although a to_boolean should yield false.

I would change this function to jerry_get_logical_value